asmcomp/power/emit.mlp

```diff
@@ -477,15 +508,21 @@ module BR = Branch_relaxation.Make (struct
     | Lop(Idls_get) -> 1
     | Lop(Ireturn_addr) -> 1
     | Lreloadretaddr -> 2
-    | Lreturn -> 2
+    | Lreturn -> 1 + if f.fun_frame_required then 1 else 0
     | Llabel _ -> 0
     | Lbranch _ -> 1
+    | Lcondbranch ((Ifloattest c), _) ->
+       let cror_s = match c with
+         | CFle | CFnle  | CFge | CFnge -> 1
+         | _ -> 0
+       in
+       1 + 1 + cror_s
     | Lcondbranch _ -> 2
     | Lcondbranch3(lbl0, lbl1, lbl2) ->
       1 + (if lbl0 = None then 0 else 1)
         + (if lbl1 = None then 0 else 1)
         + (if lbl2 = None then 0 else 1)
-    | Lswitch _ -> 7 + tocload_size
+    | Lswitch _ -> 6 + tocload_size
```

Comment:
`Lswitch` can expand to 7 instructions plus a toc load, if the number of jump tables in the current compilation unit is high enough:
https://github.com/ocaml/ocaml/blob/c82ce40504f0875969bf86b22e4d6ec7e26b3153/asmcomp/power/emit.mlp#L919-L920

Reply:
Perhaps just this `Lswitch` can be changed to check for 6 or 7 instructions plus a toc load as a compromise or emit the check if it's difficult to predict accurately.

I found enough differences in instr_size calculation to suggest the assertions are worth persisting with, though nothing in compiling the compiler libs triggered the `7` case.